bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/internal/kostal/inverter/secondgeneration/SecondGenerationHandler.java

```diff
@@ -0,0 +1,354 @@
+            channelPosts[channelPostsCounter] = dxsentries.getName();
+            channelPostsCounter++;
+        }
+
+        // Fill channelPostsExt with each item value
+        int channelPostsCounterExt = 0;
+        for (SecondGenerationDxsEntries dxsentriesExt : dxsEntriesContainerExt.dxsEntries) {
+            channelPostsExt[channelPostsCounterExt] = dxsentriesExt.getName();
+            channelPostsCounterExt++;
+        }
+
+        // Fill channelPostsExtExt with each item value
+        int channelPostsCounterExtExt = 0;
+        for (SecondGenerationDxsEntries dxsentriesExtExt : dxsEntriesContainerExtExt.dxsEntries) {
+            channelPostsExtExt[channelPostsCounterExtExt] = dxsentriesExtExt.getName();
+            channelPostsCounterExtExt++;
+        }
+
+        // Create and update actual values for each channelPost
+        int channelValuesCounter = 0;
+
+        for (SecondGenerationChannelConfiguration cConfig : channelConfigs) {
+            Channel channel = getThing().getChannel(cConfig.id);
+            State state = getState(channelPosts[channelValuesCounter], cConfig.unit);
+
+            // Update the channel
+            if (state != null) {
+                updateState(channel.getUID().getId(), state);
+                channelValuesCounter++;
+            }
+        }
+
+        // Create and update actual values for each channelPostExt
+        int channelValuesCounterExt = 0;
+        for (SecondGenerationChannelConfiguration cConfig : channelConfigsExt) {
+            Channel channel = getThing().getChannel(cConfig.id);
+            State state = getState(channelPostsExt[channelValuesCounterExt], cConfig.unit);
+
+            // Update the channel
+            if (state != null) {
+                updateState(channel.getUID().getId(), state);
+                channelValuesCounterExt++;
+            }
+        }
+
+        // Create and update actual values for each channelPostExtExt
+        int channelValuesCounterExtExt = 0;
+        for (SecondGenerationChannelConfiguration cConfig : channelConfigsExtExt) {
+            Channel channel = getThing().getChannel(cConfig.id);
+            State state = getState(channelPostsExtExt[channelValuesCounterExtExt], cConfig.unit);
+
+            // Update the channel
+            if (state != null) {
+                updateState(channel.getUID().getId(), state);
+                channelValuesCounterExtExt++;
+            }
+        }
+    }
+
+    // Helper method to handleCommand method
+    public final void preSetExecuteConfigurationChanges(HttpClient httpClient, String url, String username,
+            String password, String dxsEntriesConf, String valueConfiguration) {
+        try {
+            SecondGenerationConfigurationHandler.executeConfigurationChanges(httpClient, url, username, password,
+                    dxsEntriesConf, valueConfiguration);
+        } catch (Exception handleCommandException) {
```

Comment:
You should specify the concrete exception you want to catch, since catching `Exception` catches also all runtime exceptions.

Reply:
Can you comment on this?